pkg/datastore/mysql/query.go

```diff
@@ -31,3 +36,60 @@ func buildPutQuery(table string) string {
 func buildCreateQuery(table string) string {
 	return fmt.Sprintf("INSERT INTO %s (id, data) VALUE (UUID_TO_BIN(?,true), ?)", table)
 }
+
+func buildFindQuery(table string, ops datastore.ListOptions) string {
+	rawQuery := fmt.Sprintf(
+		"SELECT data FROM %s %s %s %s",
+		table,
+		buildWhereClause(ops.Filters),
+		buildOrderByClause(ops.Orders),
+		buildPaginationClause(ops.Page, ops.PageSize),
+	)
+	return strings.Join(strings.Fields(rawQuery), " ")
+}
+
+func buildWhereClause(filters []datastore.ListFilter) string {
+	if len(filters) == 0 {
+		return ""
+	}
+
+	var conds []string
+	for _, filter := range filters {
+		conds = append(conds, fmt.Sprintf("%s %s ?", filter.Field, filter.Operator))
+	}
+	return fmt.Sprintf("WHERE %s", strings.Join(conds[:], " AND "))
+}
+
+func buildOrderByClause(orders []datastore.Order) string {
+	if len(orders) == 0 {
+		return ""
+	}
+
+	var conds []string
```

Comment:
nits: Better pre-allocate.
```suggestion
	conds := make([]string, 0, len(orders))
```

Reply:
You're right 👍 Addressed by https://github.com/pipe-cd/pipe/pull/1678/commits/b0a18ae6cd330aacab382b8331196cdbea7f9223 🙏